freqtrade/data/history/history_utils.py

```diff
@@ -103,7 +103,9 @@ def load_data(datadir: Path,
                                  data_handler=data_handler,
                                  candle_type=candle_type,
                                  )
+
         if not hist.empty:
+            hist = reduce_dataframe_footprint(hist)
```

Comment:
i'm not a huge fan of putting this here (mostly, because it's non optional, but also because it changes values in a wrong way - see other comment for details).